rlwe/ciphertext.go

```diff
@@ -0,0 +1,335 @@
+package rlwe
+
+import (
+	"fmt"
+
+	"github.com/tuneinsight/lattigo/v4/ring"
+	"github.com/tuneinsight/lattigo/v4/rlwe/ringqp"
+	"github.com/tuneinsight/lattigo/v4/utils"
+)
+
+// CiphertextQP is a generic type for RLWE ciphertexts in R_qp.
+type CiphertextQP struct {
+	Value [2]ringqp.Poly
+}
+
+func (el *CiphertextQP) CopyNew() *CiphertextQP {
+	return &CiphertextQP{Value: [2]ringqp.Poly{el.Value[0].CopyNew(), el.Value[1].CopyNew()}}
+}
+
+// Ciphertext is a generic type for RLWE ciphertexts.
+type Ciphertext struct {
+	Value []*ring.Poly
+	*Scale
```

Comment:
possibility: we could have this field as `Scale Scale` and have the zero value of the `Scale` type be equivalent a scale of `1.0`.

Reply:
Done by having a default scale in the `rlwe.Parameters`